tools/odbc/src/statement/statement_functions.cpp

```diff
@@ -465,12 +474,21 @@ SQLRETURN duckdb::GetDataStmtResult(OdbcHandleStmt *hstmt, SQLUSMALLINT col_or_p
 		}
 		return SQL_SUCCESS;
 	}
+	// https://docs.microsoft.com/en-us/sql/odbc/reference/develop-app/datetime-data-type-changes
+	case SQL_C_TIME:
 	case SQL_C_TYPE_TIME: {
 		dtime_t time;
 		switch (val.type().id()) {
 		case LogicalTypeId::TIME:
 			time = val.GetValue<dtime_t>();
 			break;
+		case LogicalTypeId::TIME_TZ:
+		case LogicalTypeId::TIMESTAMP_TZ:
+			if (!val.TryCastAs(*hstmt->dbc->conn->context, duckdb::LogicalType::TIME, true)) {
+				return ThrowInvalidCast("GetDataStmtResult", val.type(), LogicalType::TIME, hstmt);
```

Comment:
Instead of throwing an invalid cast could you populate a diagnostic record here instead and return SQL_ERROR?